### ISSUE TITLE
GetRawValue 

### ISSUE THREAD

**Issue by ranjithpp** [Tencent/rapidjson#1659]
is there an implementation exist while DOM Parse or SAX reader to get the 'RawJson' if the type is string?
eg: {"name":"RawJson"}, 
if this parsed, GetString will give RawJson. is there a possibility to get "RawJson"? requirement is to store the json as is in database.

**Comment by ranjithpp** [comment]
i saw a RawValue for SAX writer and also a RawNumber if the stringtoNumber flag is enabled. i am thinking to modify and to call GetRaWValue() if a specificflag is enabled.

**Comment by miloyip** [comment]
I think it is not possible if "raw" means any JSON.
For example, `"` in a JSON needs to be escaped if it is stored as a string in another JSON.

**Comment by ranjithpp** [comment]
#include "rapidjson/reader.h"
#include <iostream>
#include "rapidjson/document.h"     // rapidjson's DOM-style API
#include "rapidjson/prettywriter.h" // for stringify JSON
#include "rapidjson/filereadstream.h"
#include <string>
#include <cstdio>


using namespace rapidjson;

int main()
{
          FILE* fp = fopen("foo.json", "r");
          char readBuffer[65536];
          FileReadStream is (fp, readBuffer, sizeof(readBuffer));
// ...
Document document;
//document.Parse<kParseEscapeSequenceFlag>(readBuffer);
document.Parse(readBuffer);
printf("hello = %s\n", document["hello"].GetString());
}
# cat foo.json
{ "hello":"\"John\"", "age":30, "car":null }
hasMember hello gives the below output, but the requirement is to get "\"John\"". GetString will strip and give below, but when SetString is called not sure it will be escaped and added.
 ./dom
hello = "John"
After making the changes in reader.h
# ./dom
hello = "\"John\""
#

**Comment by miloyip** [comment]
No. It is not possible if the string is not escaped. `{ "hello":""John"", "age":30, "car":null }` is not a valid JSON. The parser cannot determine where is the end of the string after `{"hello":"`.

**Comment by ranjithpp** [comment]
looks like the formatter in github skipped the escape sequence!! intention is "\\\"John\\\""

**Comment by ranjithpp** [comment]
i may need to rephrase the question, requirement is to get the json representation of a string. 
if GetString is called for "Test" Test is the value, instead "Test" to be returned.

**Comment by miloyip** [comment]
Can you just use writer to make the string as a JSON again?

**Comment by ranjithpp** [comment]
https://github.com/Tencent/rapidjson/issues/1230 i started from here. is there an option available in Dom?

**Comment by ranjithpp** [comment]
can i do a pull request to make necessary changes ?
